sycl/include/CL/sycl/detail/cg.hpp

```diff
@@ -45,11 +47,12 @@ class ArgDesc {
 class NDRDescT {
   // The method initializes all sizes for dimensions greater than the passed one
   // to the default values, so they will not affect execution.
-  template <int Dims_> void setNDRangeLeftover() {
+  void setNDRangeLeftover(int Dims_) {
     for (int I = Dims_; I < 3; ++I) {
       GlobalSize[I] = 1;
       LocalSize[I] = LocalSize[0] ? 1 : 0;
       GlobalOffset[I] = 0;
+      NumWorkGroups[I] = 0;
```

Comment:
I wonder how the compiler will optimize all these redundant allocations in the case of Dims = 1 or 2, instead of having exactly the right size...

Reply:
Right, there maybe some small inefficiency, but it is host-side only and we "buy" independence of template,  as @romanovvlad commented in the original review. Do you think we need to file an issue on this?